### PR TITLE
Add how to run tests and lint check for MLfow UI in `CONTRIBUTING.rst`

### PR DESCRIPTION
**Pull request by ahlag**
Signed-off-by: changyonglik <theeahlag@gmail.com>

## Related Issues/PRs
Added more details about testing React components

N/A

## What changes are proposed in this pull request?
Added a Testing and Linting Sections to CONTRIBUTING.rst
(Please fill in changes proposed in this fix)

## How is this patch tested?
N/A

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [x] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -279,6 +279,41 @@ In another shell:
 
 The MLflow Tracking UI will show runs logged in ``./mlruns`` at `<http://localhost:3000>`_.
 
+Testing React Component
```
Comment (dbczumar) [inline CONTRIBUTING.rst:282]:
```suggestion
Testing a React Component
```

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @ahlag ! Can you follow the instructions in https://github.com/mlflow/mlflow/pull/6330/checks?check_run_id=7519132885 to sign your commits before merging?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -279,6 +279,41 @@ In another shell:
...
++++++++++++++++++++++++++++++++++
```
Comment (harupy) [inline CONTRIBUTING.rst:283]:
```suggestion
+++++++++++++
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -279,6 +279,41 @@ In another shell:
...
+   › Press f to run only failed tests.
+   › Press q to quit watch mode.
+   › Press p to filter by a filename regex pattern.
+   › Press t to filter by a test name regex pattern.
+   › Press Enter to trigger a test run.
+
+Press ``a`` to run all the tests or ``q`` to quit.
+
+Linting Javascript Code
++++++++++++++++++++++++++++++++++
```
Comment (harupy) [inline CONTRIBUTING.rst:310]:
```suggestion
+++++++++++++
```

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -279,6 +279,41 @@ In another shell:
...
+.. code-block:: bash
+
+   yarn test
+
+Once the test is loaded, you will be prompted with an option menu as below:
+
+.. code-block:: bash
+
+  No tests found related to files changed since last commit.
+  Press `a` to run all tests, or run Jest with `--watchAll`.
```
Comment (harupy) [inline CONTRIBUTING.rst:297]: `yarn test` executes `yarn test:base --watchAll=false` which should run tests in non-interactive mode.

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -279,6 +279,41 @@ In another shell:
...
 
+Testing React Component
++++++++++++++++++++++++++++++++++
+Add a test component in the same directory as the newly created React component. 
+For example, ``CompareRunBox.test.js`` should be added in the same directory as ``CompareRunBox.js``.
+Next, in ``mlflow/server/js``, run the following command to start the test.
+
+.. code-block:: bash
+
+   yarn test
```
Comment (harupy) [inline CONTRIBUTING.rst:290]:
```suggestion
   # Run tests in CompareRunBox.test.js
   yarn test CompareRunBox.test.js
   # Run 'click plot' test in CompareRunBox.test.js
   yarn test CompareRunBox.test.js -t 'click plot'
   # Run all tests
   yarn test  # run all tests
```

Let's add a few more commands.

File: CONTRIBUTING.rst (hunk 4 above)
Comment (harupy) [reply CONTRIBUTING.rst:297]: On my machine, `yarn test` runs tests in non-interactive mode.

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -279,6 +279,41 @@ In another shell:
...
+
+Press ``a`` to run all the tests or ``q`` to quit.
+
+Linting Javascript Code
++++++++++++++++++++++++++++++++++
+In ``mlflow/server/js``, run the following command to lint your code.
+
+.. code-block:: bash
+
+   yarn lint:fix
```
Comment (harupy) [inline CONTRIBUTING.rst:315]:
```suggestion
   # Note this command only fixes auto-fixable issues (e.g. remove trailing whitespace)
   yarn lint:fix
```

File: CONTRIBUTING.rst (hunk 5 above)
Comment (ahlag) [reply CONTRIBUTING.rst:290]: Ok thanks!

File: CONTRIBUTING.rst (hunk 4 above)
Comment (ahlag) [reply CONTRIBUTING.rst:297]: That seems strange. On my mac, it ran interactively. I will change the wording to yours.

**Comment by ahlag** [comment]
@dbczumar 
Hey thanks for the review sure!

@harupy 
I have fixed the doc.

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -279,6 +279,32 @@ In another shell:
...
+  yarn test CompareRunBox.test.js
+  # Run tests with a name that matches 'plot' in CompareRunBox.test.js
+  yarn test CompareRunBox.test.js -t 'plot'
+  # Run all tests
+  yarn test
+
+``yarn test`` executes ``yarn test:base --watchAll=false``, which should run tests in non-interactive mode.
+
+Linting Javascript Code
++++++++++++++
```
Comment (harupy) [inline CONTRIBUTING.rst:300]:
```suggestion
Linting Javascript Code
+++++++++++++++++++++++
```

Sorry, the number of `+` was incorrect in my previous suggestion.

**Comment by ahlag** [comment]
@harupy 
Ok done!

File: CONTRIBUTING.rst [hunk 8]
```diff
@@ -279,6 +279,32 @@ In another shell:
 
 The MLflow Tracking UI will show runs logged in ``./mlruns`` at `<http://localhost:3000>`_.
 
+Testing a React Component
++++++++++++++
```
Comment (harupy) [inline CONTRIBUTING.rst:283]: Could you fix this line?

File: CONTRIBUTING.rst [hunk 9]
```diff
@@ -279,6 +279,32 @@ In another shell:
...
+Add a test component in the same directory as the newly created React component.
```
Comment (harupy) [inline CONTRIBUTING.rst:284]:
```suggestion
Add a test file in the same directory as the newly created React component.
```

Comment (ahlag) [reply CONTRIBUTING.rst]: Ok done!

File: CONTRIBUTING.rst [hunk 10]
```diff
@@ -279,6 +279,32 @@ In another shell:
 
 The MLflow Tracking UI will show runs logged in ``./mlruns`` at `<http://localhost:3000>`_.
 
+Testing a React Component
++++++++++++++++++++++++
```
Comment (harupy) [inline CONTRIBUTING.rst:283]:
```suggestion
+++++++++++++++++++++++++
```

File: CONTRIBUTING.rst [hunk 11]
```diff
@@ -279,6 +279,32 @@ In another shell:
...
+.. code-block:: bash
+
+  # Run tests in CompareRunBox.test.js
+  yarn test CompareRunBox.test.js
+  # Run tests with a name that matches 'plot' in CompareRunBox.test.js
+  yarn test CompareRunBox.test.js -t 'plot'
+  # Run all tests
+  yarn test
+
+``yarn test`` executes ``yarn test:base --watchAll=false``, which should run tests in non-interactive mode.
```
Comment (harupy) [inline CONTRIBUTING.rst:297]:
```suggestion
```

I think we can remove this block.

**Comment by ahlag** [comment]
Done

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**